src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -567,6 +567,10 @@ public function mirror($originDir, $targetDir, \Traversable $iterator = null, $o
         }
 
         foreach ($iterator as $file) {
+            if ($file->getPathName() === $targetDir) {
```

Comment:
I'm not sure how/if this works with relative directories, but we can care in a later PR if anyone is up to.